File: Codes from the AI-assistants/[Experimental] pi-digits/Basic Prompt - Normal Mode/PiDigits-deepseek-basic-normal.py
from decimal import Decimal, getcontext

def compute_pi_bbp(n):
    """Compute pi to n digits using the BBP formula with Decimal arithmetic."""
    getcontext().prec = n + 2  # extra precision for intermediate steps
    
    pi = Decimal(0)
    for k in range(n + 1):
        term = (Decimal(1)/(16**k)) * (
            Decimal(4)/(8*k + 1) -
            Decimal(2)/(8*k + 4) -
            Decimal(1)/(8*k + 5) -
            Decimal(1)/(8*k + 6)
        )
        pi += term
    
    # Return pi as a string with n digits
    return str(pi)[:n+2]  # +2 for "3."

# Example usage:
n = 100  # Number of digits
pi_digits = compute_pi_bbp(n)
print(f"Pi to {n} digits: {pi_digits}")

File: Codes from the AI-assistants/[Experimental] pi-digits/Basic Prompt - Normal Mode/PiDigits-deepseek-basic-normal.java
import java.math.BigDecimal;
import java.math.MathContext;

public class PiCalculator {
    public static String computePiBBP(int digits) {
        MathContext mc = new MathContext(digits + 2); // extra precision
        BigDecimal pi = BigDecimal.ZERO;
        
        for (int k = 0; k <= digits; k++) {
            BigDecimal term1 = BigDecimal.valueOf(4).divide(
                BigDecimal.valueOf(8 * k + 1), mc);
            BigDecimal term2 = BigDecimal.valueOf(2).divide(
                BigDecimal.valueOf(8 * k + 4), mc);
            BigDecimal term3 = BigDecimal.valueOf(1).divide(
                BigDecimal.valueOf(8 * k + 5), mc);
            BigDecimal term4 = BigDecimal.valueOf(1).divide(
                BigDecimal.valueOf(8 * k + 6), mc);
            
            BigDecimal term = term1.subtract(term2)
                                 .subtract(term3)
                                 .subtract(term4)
                                 .divide(BigDecimal.valueOf(16).pow(k), mc);
            
            pi = pi.add(term);
        }
        
        return pi.toString().substring(0, digits + 2); // +2 for "3."
    }

    public static void main(String[] args) {
        int n = 100; // Number of digits
        String piDigits = computePiBBP(n);
        System.out.println("Pi to " + n + " digits: " + piDigits);
    }
}